core/trino-spi/src/main/java/io/trino/spi/shuffle/Shuffle.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.shuffle;
+
+import java.io.Closeable;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+public interface Shuffle
```

Comment:
The naming of this interface should probably have `input` and `output` reversed, as it seems that output refers to the output of the mapper, input refers to the input of reducers. This is a little confusing, as the output of the mapper is the input to the shuffle, and the input of the reducer is the output of the shuffle

Reply:
Updated the SPI based on the offline discussion